opsdroid/connector/matrix/connector.py

```diff
@@ -133,8 +141,22 @@ async def make_filter(self, api, fjson):
 
     async def connect(self):
         """Create connection object with chat library."""
-        config = nio.AsyncClientConfig(encryption_enabled=False, pickle_key="")
-        mapi = nio.AsyncClient(self.homeserver, self.mxid, config=config)
+
+        if self._allow_encryption and not Path(self.store_path).is_dir():
+            Path(self.store_path).mkdir()
+
+        config = nio.AsyncClientConfig(
+            encryption_enabled=self._allow_encryption,
+            pickle_key="",
+            store_name="opsdroid.db" if self._allow_encryption else "",
```

Comment:
does this store have any value if e2ee isn't enabled?

Reply:
nope it doesn't get created at all

Reply:
then do we need this conditional?

Reply:
if we pass a non-empty string it tries to load the store which then errors out

Reply:
Only if encryption is disabled?

Reply:
yeah